### PR TITLE
Don't override user set port

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

**Comment by mbogoevici** [comment]
This still fails to undeploy a module that specifies a custom port, e.g.:

```
dataflow:>stream create httptock --definition "http --server.port=9090 | log" --deploy
Created and deployed new stream 'httptock'
dataflow:>http post --target http://localhost:9090/messages --data "Hi"
> POST (text/plain;Charset=UTF-8) http://localhost:9090/messages Hi
> 202 ACCEPTED

dataflow:>http post --target http://localhost:9090/messages --data "HiHi"
> POST (text/plain;Charset=UTF-8) http://localhost:9090/messages HiHi
> 202 ACCEPTED

dataflow:>stream destroy httptock
Command failed org.springframework.cloud.dataflow.rest.client.DataFlowClientException: I/O error on POST request for "http://192.168.0.14:19815/shutdown":Connection refused; nested exception is java.net.ConnectException: Connection refused
```

File: spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java [hunk 1]
```diff
@@ -90,6 +90,7 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 		args.put("endpoints.shutdown.enabled", "true");
 		args.put("endpoints.jmx.unique-names", "true");
 
+		boolean useDynamicPort = !args.containsKey(SERVER_PORT_KEY);
```
Comment (mbogoevici) [inline spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java]: At this point, args are already qualified, so this is always false.

File: spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java [hunk 2]
```diff
@@ -98,8 +99,10 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 				workDir.toFile().deleteOnExit();
 			}
 			for (int i = 0; i < request.getCount(); i++) {
-				int port = SocketUtils.findAvailableTcpPort(DEFAULT_SERVER_PORT);
-				args.put(SERVER_PORT_KEY, String.valueOf(port));
+				if (useDynamicPort) {
+					int port = SocketUtils.findAvailableTcpPort(DEFAULT_SERVER_PORT);
+					args.put(SERVER_PORT_KEY, String.valueOf(port));
```
Comment (mbogoevici) [inline spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java]: The qualified value should be put here.

**Comment by ericbottard** [comment]
Fixed and tested

**Comment by mbogoevici** [comment]
Rebased, squashed and merged w/https://github.com/spring-cloud/spring-cloud-dataflow/commit/27ca7cb127007f70a686c23811795f34fd68847e

**Pull request closed without merging**